crates/tests/src/lib.rs

```diff
@@ -252,3 +260,195 @@ async fn madara_can_sync_a_few_blocks() {
+            calldata: vec![ACCOUNT_ADDRESS, 15.into(), Felt::ZERO],
+        }])
+        .send()
+        .await
+        .unwrap();
+
+    drop(node);
+
+    // tx should be in saved mempool
+
+    let cmd_builder = cmd_builder.args([
+        "--devnet",
+        "--no-l1-sync",
+        "--gas-price",
+        "0",
+        // never produce blocks but produce pending txs
+        "--chain-config-path",
+        "test_devnet.yaml",
+        "--chain-config-override",
+        "block_time=5min,pending_block_update_time=500ms",
+    ]);
+    let mut node = cmd_builder.clone().run();
+    node.wait_for_ready().await;
+
+    // tx should be in mempool
+
+    wait_for_cond(
+        || async {
+            let _receipt = node.json_rpc().get_transaction_receipt(res.transaction_hash).await?;
+            Ok(())
+        },
+        Duration::from_secs(1),
+    )
+    .await;
+}
+
+#[rstest]
+#[tokio::test]
+async fn madara_devnet_continue_pending() {
+    let _ = tracing_subscriber::fmt().with_test_writer().try_init();
+
+    let cmd_builder = MadaraCmdBuilder::new().args([
+        "--devnet",
+        "--no-l1-sync",
+        "--gas-price",
+        "0",
+        // never produce blocks but produce pending txs
+        "--chain-config-path",
+        "test_devnet.yaml",
+        "--chain-config-override",
+        "block_time=5min,pending_block_update_time=500ms",
+    ]);
+    let mut node = cmd_builder.clone().run();
+    node.wait_for_ready().await;
+
+    let chain_id = node.json_rpc().chain_id().await.unwrap();
+
+    let signer = LocalWallet::from_signing_key(SigningKey::from_secret_scalar(ACCOUNT_SECRET));
+    let mut account =
+        SingleOwnerAccount::new(node.json_rpc(), signer, ACCOUNT_ADDRESS, chain_id, ExecutionEncoding::New);
+    account.set_block_id(BlockId::Tag(BlockTag::Pending));
+
+    let res = account
+        .execute_v3(vec![Call {
+            to: ERC20_STRK_CONTRACT_ADDRESS,
+            selector: starknet_keccak(b"transfer"),
+            calldata: vec![ACCOUNT_ADDRESS, 15.into(), Felt::ZERO],
+        }])
+        .send()
+        .await
+        .unwrap();
+
+    wait_for_cond(
+        || async {
+            let _receipt = node.json_rpc().get_transaction_receipt(res.transaction_hash).await?;
+            Ok(())
+        },
+        Duration::from_secs(1),
+    )
+    .await;
+
+    drop(node);
+
+    // tx should appear in saved pending block
+
+    let cmd_builder = cmd_builder.args([
+        "--devnet",
+        "--no-l1-sync",
+        "--gas-price",
+        "0",
+        // never produce blocks never produce pending txs
+        "--chain-config-path",
+        "test_devnet.yaml",
+        "--chain-config-override",
+        "block_time=5min,pending_block_update_time=5min",
+    ]);
+    let mut node = cmd_builder.clone().run();
+    node.wait_for_ready().await;
+
+    // should find receipt
+    let _receipt = node.json_rpc().get_transaction_receipt(res.transaction_hash).await.unwrap();
```

Comment:
Maybe replace this `unwrap` with an `assert`?

Reply:
i think it would look more janky, what do you have in mind?
assert result is_ok()? that wouldnt show the error to the console if the assertion fails

Reply:
How about an `expect`?

Reply:
it's in a test, i usually never use expect in tests and just use unwraps
I think it's fine to unwrap in tests